src/pipeline_migration/actions/migrate/__init__.py

```diff
@@ -0,0 +1,58 @@
+# Expose the main entry point and CLI registration from main.py
+from .main import migrate, register_cli, clean_upgrades
+
+# Expose core data classes and exceptions
+from .core import (
+    ANNOTATION_HAS_MIGRATION,
+    ANNOTATION_IS_MIGRATION,
+    ANNOTATION_TRUTH_VALUE,
+    MIGRATION_IMAGE_TAG_LIKE_PATTERN,
+    ANNOTATION_PREVIOUS_MIGRATION_BUNDLE,
+    MigrationImageTag,
+    TaskBundleUpgrade,
+    TaskBundleMigration,
+    PackageFile,
+    InvalidRenovateUpgradesData,
+    MigrationResolveError,
+    MigrationApplyError,
+    IncorrectMigrationAttachment,
+    determine_task_bundle_upgrades_range,
+    fetch_migration_file,
+    list_bundle_tags,
+    drop_out_of_order_versions,
+)
+
+# Expose resolver classes for type hinting and direct use
+from .resolver import Resolver
+from .simple_iteration_resolver import SimpleIterationResolver
+from .linked_migrations_resolver import LinkedMigrationsResolver
+from .migration_images_resolver import MigrationImagesResolver
+from .resolver_proxy import DecentralizationTransitionResolverProxy
+
+__all__ = [
+    "migrate",
+    "register_cli",
+    "clean_upgrades",
+    "ANNOTATION_HAS_MIGRATION",
+    "ANNOTATION_IS_MIGRATION",
+    "ANNOTATION_TRUTH_VALUE",
+    "MIGRATION_IMAGE_TAG_LIKE_PATTERN",
+    "ANNOTATION_PREVIOUS_MIGRATION_BUNDLE",
+    "MigrationImageTag",
+    "TaskBundleUpgrade",
+    "TaskBundleMigration",
+    "PackageFile",
+    "InvalidRenovateUpgradesData",
+    "MigrationResolveError",
+    "MigrationApplyError",
+    "IncorrectMigrationAttachment",
+    "determine_task_bundle_upgrades_range",
+    "fetch_migration_file",
+    "list_bundle_tags",
+    "drop_out_of_order_versions",
+    "Resolver",
+    "SimpleIterationResolver",
+    "LinkedMigrationsResolver",
+    "MigrationImagesResolver",
+    "DecentralizationTransitionResolverProxy",
+]
```

Comment:
So far, `__all__` is useless. We do not import by:

```python
from pipeline_migration.actions.migrate import *
```

which is also not a recommended way to import stuff from a package generally.

Instead, import names from specific modules. Please update the tests.